pkg/controllers/restore/controller.go

```diff
@@ -626,6 +626,24 @@ func (h *handler) restoreResource(restoreObjInfo objInfo, restoreObjData unstruc
 			}
 		}
 	}
+	// Check for invalid v1beta1 fields if the APIVersion is v1
+	if obj.GetAPIVersion() == "apiextensions.k8s.io/v1" {
+		// Invalid field is spec.preserveUnknownFields
+		if _, ok := obj.Object["spec"].(map[string]interface{})["preserveUnknownFields"]; ok {
```

Comment:
Can we change "preserveUnknownFields" to a const string - Also if there are multiple such v1beta1 fields in the spec, it might be a good idea to store them in a map and have a utility method that checks if any of these v1beta1 fields is present in the spec fields and returns back the fields found. what do you think? T

Reply:
Yes that sounds good but we don't have any other fields that we need to check or correct currently, maybe when it comes back from testing but I already ran tests for this on 2.6.0 -> 2.6.3 and did not occur any issue with this fix in place.

Reply:
sure, we can add this in future if any more fields across new apiversions face the same problem.